docker-compose.yaml

```diff
@@ -0,0 +1,83 @@
+version: '2'
+
+services:
+
+  main:
+    build:
+      context: .
+      dockerfile: Dockerfile
+      args:
+        - http_proxy
+        - https_proxy
+        - no_proxy
+        - BUILDDOC=OFF
+        - BUILDSWIG=OFF
+        - BUILDSWIGPYTHON=OFF
+        - BUILDSWIGJAVA=OFF
+        - BUILDSWIGNODE=OFF
+        - USBPLAT=OFF
+        - FIRMATA=OFF
+        - ONEWIRE=OFF
+        - JSONPLAT
+        - IMRAA=OFF
+        - FTDI4222=OFF
+        - IPK=OFF
+        - RPM=OFF
+        - ENABLEEXAMPLES=ON
+        - INSTALLGPIOTOOL=OFF
+        - INSTALLTOOLS=OFF
+        - BUILDTESTS=ON
+        - CC=clang-3.8
+        - CXX=clang++-3.8
+        - NODE_VERSION=v4.1.2
+        - BUILDARCH
+    environment:
+      - http_proxy
+      - https_proxy
+      - no_proxy
+
+  doc:
+    extends: main
+    build:
+      args:
+        - BUILDDOC=ON
+    command: make doc
```

Comment:
> It could look better if every command just use bash -c everywhere, thought

+1 for that, would avoid unnecessary confusion several months from now when no one remembers why everything has `bash -c` and this one doesn't :)